mithril-aggregator/src/services/signable_seed_builder.rs

```diff
@@ -0,0 +1,85 @@
+//! ## SignableSeedBuilderService
+//!
+//! This service is responsible for computing the seed protocol message
+//! that is used by the [SignableBuilder] to compute the final protocol message.
+//!
+use anyhow::Context;
+use async_trait::async_trait;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+use mithril_common::{
+    entities::{ProtocolMessage, ProtocolMessagePartKey},
+    signable_builder::SignableSeedBuilder,
+    StdResult,
+};
+
+use crate::services::EpochService;
+
+/// SignableSeedBuilder service
+pub struct SignableSeedBuilderService {
+    epoch_service: Arc<RwLock<dyn EpochService>>,
+}
+
+impl SignableSeedBuilderService {
+    /// SignableSeedBuilderService factory
+    pub fn new(epoch_service: Arc<RwLock<dyn EpochService>>) -> Self {
+        Self { epoch_service }
+    }
+}
+
+#[async_trait]
+impl SignableSeedBuilder for SignableSeedBuilderService {
+    /// Compute seed protocol message
+    async fn compute_seed_protocol_message(&self) -> StdResult<ProtocolMessage> {
+        let epoch_service = self.epoch_service.read().await;
+        let mut protocol_message = ProtocolMessage::new();
```

Comment:
I wonder if this service should only return a ProtocolMessagePartValue and let the building of ProtocolMessage to the SignableBuilderService.
This would avoid to return a ProtocolMessage that is not fully completed.

Reply:
Done, see https://github.com/input-output-hk/mithril/pull/1942#discussion_r1773415634 :+1: